packages/core/src/State.ts

```diff
@@ -257,7 +256,7 @@ export class State<
   /**
    * The next events that will cause a transition from the current state.
    */
-  public get nextEvents(): Array<EventDescriptor<TEvent>> {
+  public get nextEvents(): Array<TEvent['type']> {
```

Comment:
This change was not correct and I quickly force-pushed the previous version of `next` once this was merged in. Please share a test case demoing your problems so I can help with your exact problem.